pkg/metadata/metadata.go

```diff
@@ -86,3 +89,51 @@ func ApplyAnnotations(c client.Client, ctx context.Context, node core.Node, anno
 	}
 	return nil
 }
+
+// ApplyVersionAnnotation applies this operator's version as the version annotation to the given Node
+func ApplyVersionAnnotation(ctx context.Context, c client.Client, node core.Node, value string) error {
+	return ApplyLabelsAndAnnotations(ctx, c, node, nil, map[string]string{VersionAnnotation: value})
+}
+
+// ApplyDesiredVersionAnnotation applies this operator's version as the desired version annotation to the given Node
+func ApplyDesiredVersionAnnotation(ctx context.Context, c client.Client, node core.Node) error {
+	return ApplyLabelsAndAnnotations(ctx, c, node, nil, map[string]string{DesiredVersionAnnotation: version.Get()})
+}
+
+// RemoveVersionAnnotation clears the version annotation from the node object, indicating the node is not configured
+func RemoveVersionAnnotation(ctx context.Context, c client.Client, node core.Node) error {
+	if _, present := node.GetAnnotations()[VersionAnnotation]; present {
+		patchData, err := GenerateRemovePatch([]string{}, []string{VersionAnnotation})
+		if err != nil {
+			return errors.Wrapf(err, "error creating version annotation remove request")
+		}
+		err = c.Patch(ctx, &node, client.RawPatch(kubeTypes.JSONPatchType, patchData))
+		if err != nil {
+			return errors.Wrapf(err, "error removing version annotation from node %s", node.GetName())
+		}
+	}
+	return nil
+}
+
+// WaitForVersionAnnotation checks if the node object has equivalent version and desiredVersion annotations.
+// Waits for retry.Interval seconds and returns an error if the version annotation does not appear in that time frame.
+func WaitForVersionAnnotation(ctx context.Context, c client.Client, nodeName string) error {
+	var found bool
+	node := &core.Node{}
+	err := wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
```

Comment:
Checking  every 15 seconds for the version annotation seems very aggressive. And is a 10 minute timeout sufficient? I suggest we introduce specific constants for this case.

Reply:
In my testing I saw successful version annotation application within 2 minutes. The 10 min timeout should be fine IMO. Should I change this to poll every 30 seconds instead?

Reply:
Given what you are seeing 10 minutes is good. And if it is 2 minutes total time I am fine with 15 min intervals also.

Reply:
Looks like the WICD work has sped up configuration times significantly. Machine-backed node took less than 2 min :rocket: 
```
20:34:19 - init SSH connection 
20:35:58 - start WICD controller
20:36:36 - kubelet posts ready status
20:37:13 - instance has been fully configured as a worker node
```

Reply:
I expect we can make this even faster by moving the HNS network removal to WICD as part of it's clean up or a separate sub-command.